keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/vetos/RequiredSecurityGroupVeto.kt

```diff
@@ -0,0 +1,115 @@
+package com.netflix.spinnaker.keel.ec2.vetos
+
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceSpec
+import com.netflix.spinnaker.keel.api.ec2.LoadBalancerSpec
+import com.netflix.spinnaker.keel.api.ec2.OverrideableClusterDependencyContainer
+import com.netflix.spinnaker.keel.api.ec2.securityGroupsByRegion
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverCache
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverService
+import com.netflix.spinnaker.keel.core.api.DEFAULT_SERVICE_ACCOUNT
+import com.netflix.spinnaker.keel.ec2.CLOUD_PROVIDER
+import com.netflix.spinnaker.keel.veto.Veto
+import com.netflix.spinnaker.keel.veto.VetoResponse
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.supervisorScope
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import retrofit2.HttpException
+
+@Component
+class RequiredSecurityGroupVeto(
+  private val cloudDriver: CloudDriverService,
+  private val cloudDriverCache: CloudDriverCache
+) : Veto {
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  override suspend fun check(resource: Resource<*>) =
+    resource.spec.toSecurityGroupDependencies()
+      ?.let {
+        val missingSecurityGroupRegions = checkSecurityGroups(it)
+        VetoResponse(
+          allowed = missingSecurityGroupRegions.isEmpty(),
+          vetoName = name(),
+          message = missingSecurityGroupRegions.entries.joinToString(separator = "\n") { (securityGroup, missingRegions) ->
+            "Security group $securityGroup is not found in ${missingRegions.joinToString()}"
```

Comment:
Did you want to separate the regions with commas here?

Reply:
Default separator on `joinToString` is `", "` so it will